cmd/webhook/admission/routegroup.go

```diff
@@ -0,0 +1,51 @@
+package admission
+
+import (
+	"encoding/json"
+
+	"github.com/zalando/skipper/dataclients/kubernetes/definitions"
+)
+
+type RouteGroupAdmitter struct {
+	RouteGroupValidator *definitions.RouteGroupValidator
+}
+
+func (rga *RouteGroupAdmitter) name() string {
+	return "routegroup"
+}
+
+func (rga *RouteGroupAdmitter) admit(req *admissionRequest) (*admissionResponse, error) {
+
+	// Serve as default validator if not set
+	if rga.RouteGroupValidator == nil {
+		rga.RouteGroupValidator = &definitions.RouteGroupValidator{}
+	}
+
+	rgItem := definitions.RouteGroupItem{}
+	err := json.Unmarshal(req.Object, &rgItem)
+	if err != nil {
+		return &admissionResponse{
+			UID:     req.UID,
+			Allowed: false,
+			Result: &status{
+				Message: err.Error(),
+			},
+		}, err
```

Comment:
We never returned error here with response, so metrics for rejected request was wrong.

See https://github.com/zalando/skipper/blob/master/cmd/webhook/admission/admission.go#L161-L167

Reply:
Generally we either should return either nil *admissionResponse or nil error. but never both nil or both set. On the caller side it is always
```go
foo, err := bar()
if err != nil {
  // handle error and return
}
// handle foo
```

Reply:
In this case and below I think it is valid to return nil error since admissionResponse would not allow and would contain error message.

Reply:
I don't need the response here yes thanks for the hint, it's also already handled in the Handler